cpp/src/arrow/filesystem/gcsfs_test.cc

```diff
@@ -604,6 +650,7 @@ TEST_F(GcsIntegrationTest, DeleteDirContentsSuccess) {
   arrow::fs::AssertFileInfo(fs.get(), PreexistingBucketPath(), FileType::Directory);
   arrow::fs::AssertFileInfo(fs.get(), PreexistingObjectPath(), FileType::File);
   for (auto const& info : hierarchy.contents) {
+    if (!fs::internal::IsAncestorOf(hierarchy.base_dir, info.path())) continue;
```

Comment:
Can you do the same change as well here? (check the expected type)

Reply:
Fixed, sorry about that.